torch/lib/THD/base/data_channels/DataChannelGloo.cpp

```diff
@@ -268,7 +277,10 @@ void DataChannelGloo::barrier(THDGroup group_id) {
   RETURN_IF_NOT_IN_GROUP
   auto ret = _cache->getAlgorithm<CollectiveType::BARRIER, void>(
     group_id, _groups.at(group_id));
-  std::get<0>(ret)->run();
+  {
+    std::lock_guard<std::mutex> lock(*std::get<3>(ret));
```

Comment:
ugh can we please define an `inline` accessor for this? `std::get` is unmaintainable